source/JsMaterialX/README.md

```diff
@@ -184,6 +184,54 @@ Since the module itself is ES5 code, we recommend to write custom code in ES5 as
 
 In order to avoid conflicting definitions in post.js files, we recommend to wrap custom code in an [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) (Immediately Invoked Function Expression).
 
+### Using the EsslShaderGenerator JavaScript bindings
+#### Setup
+Make sure to consume `JsMaterialXGenShader.js` instead of `JsMaterialXCore.js` as described here [here](#consuming-the-module). Additionally, ensure that the app serves `JsMaterialXGenShader.data` and `JsMaterialXGenShader.wasm`. The `.data` file includes the prepacked library files containing the shader snipptets and MaterialX node definitions/implementations required for generating the shader code.
+
+#### Generating Essl Shader Code & Compiling with WebGL
+To generate WebGL 2 compatible shader code a generator context and an instance of the `EsslShaderGenerator` class is required. 
+```javascript
+const gen = new mx.EsslShaderGenerator();
+const genContext = new mx.GenContext(gen);
+```
+The standard libraries need to be loaded and imported into the document. This step is required as the standard libraries contain all the definitions and snippests needed for assembly of the shader code.
+```javascript
+const stdlib = mx.loadStandardLibraries(genContext);
+const doc = mx.createDocument();
+doc.importLibrary(stdlib);
+```
+Now it is either time to load a document from a file as outline [here](loading-materialx-files) or create one using the API.
+Generating the code consists of finding a renderable element (the javascript binding returns the first renderable element) and calling the `generate` method from the `EsslShaderGenerator` class. The shader code for the vertex/pixel shader may be requested from the resulting `Shader` instance.
+```javascript
+const elem = mx.findRenderableElement(doc);
```

Comment:
What's the reason for providing a custom function that only returns one element? This way, users cannot generate shaders for all materials in a file. Why not simply use the existing `findRenderableElements` and return an array of elements?

Reply:
@sdunkel

Reply:
This is just a simplification of the code used in MaterialXViewer (see https://github.com/autodesk-forks/MaterialX/blob/adsk_contrib/dev/source/MaterialXView/Viewer.cpp#L1235). There is no dedicated function for that piece of code at the moment. 

We could extent this to multiple elements, but each would need to get their own material + shaders etc. and we would need to figure out how to map it to different parts of the geometry.
Ultimately this should probably be rewritten in an real viewing application in JS

Reply:
Yes, that's what I thought. It seems like we added an application-side convenience function to the library, and in particular one that is special to the viewer app.

I guess the intended way of getting renderable elements (from the actual library point of view) is to call `findRenderableElements` (https://github.com/autodesk-forks/MaterialX/blob/adsk_contrib/dev/source/MaterialXGenShader/Util.h#L63), which fills (or, in JS, returns) a vector/array of elements. If an application can only handle a single one, it would simply use the first one.

Reply:
This discussions is unrelated to this PR, of course. But I think we should go with an approach that is in line with the official API, instead of demoing and recommending the use of an arbitrary convenience method that shouldn't be part of the library.